wpiutil/src/main/native/include/wpi/protobuf/Protobuf.h

```diff
@@ -41,6 +45,87 @@ class SmallVectorImpl;
 template <typename T>
 struct Protobuf {};
 
+class ProtoInputStream {
+ public:
+  ProtoInputStream(pb_istream_t* stream, const pb_msgdesc_t* msgDesc)
+      : m_streamMsg{stream}, m_msgDesc{msgDesc} {}
+
+  ProtoInputStream(std::span<const uint8_t> stream, const pb_msgdesc_t* msgDesc)
+      : m_streamLocal{pb_istream_from_buffer(
+            reinterpret_cast<const pb_byte_t*>(stream.data()), stream.size())},
+        m_msgDesc{msgDesc} {}
+
+  pb_istream_t* Stream() noexcept {
+    return m_streamMsg ? m_streamMsg : &m_streamLocal;
+  }
+  const pb_msgdesc_t* MsgDesc() const noexcept { return m_msgDesc; }
+
+  template <typename T>
+  bool Decode(T& msg) {
```

Comment:
These should return a FreeMsg wrapper that will free the msg on destruction. Needed for allocated things.